wafer/talks/tests/test_review_forms.py

```diff
@@ -0,0 +1,107 @@
+"""Tests for wafer.talk review form behaviour."""
+
+from django.test import TestCase, override_settings
+from django.urls import reverse
+
+from reversion import revisions
+
+from wafer.talks.models import (SUBMITTED, UNDER_CONSIDERATION,
+                                ReviewAspect, Review)
+from wafer.talks.forms import ReviewForm, make_aspect_key
+
+from wafer.tests.utils import create_user
+from wafer.talks.tests.fixtures import create_talk
+
+
+
+
+class ReviewFormTests(TestCase):
+    def setUp(self):
+        self.reviewer_a = create_user('reviewer_a', perms=('add_review',))
+        self.talk_a = create_talk('Talk A', SUBMITTED, "author_a")
+        with revisions.create_revision():
+            # Ensure we have an initial revision
+            self.talk_a.save()
+        self.aspect_1 = ReviewAspect.objects.create(name='General')
+        self.aspect_2 = ReviewAspect.objects.create(name='Other')
+
+    def _save_review(self, review, scores=('1', '1')):
+        form = ReviewForm(instance=review, talk=self.talk_a, user=self.reviewer_a,
+                       data={'notes': "Test review",
+                             make_aspect_key(self.aspect_1): scores[0],
+                             make_aspect_key(self.aspect_2): scores[1]})
+        form.is_valid()
+        with revisions.create_revision():
+            form.save()
+        return Review.objects.get(talk=self.talk_a, reviewer=self.reviewer_a)
+
+    def test_review_form(self):
+        """Test that submitting a review works"""
+        form = ReviewForm(instance=None, talk=self.talk_a, user=self.reviewer_a,
+                       data={'notes': "Test review",
+                             make_aspect_key(self.aspect_1): '1',
+                             make_aspect_key(self.aspect_2): '1'})
+        self.assertTrue(form.is_valid())
+        self.assertEqual(Review.objects.count(), 0)
+        with revisions.create_revision():
+            form.save()
+        self.assertEqual(Review.objects.count(), 1)
+
+    def test_review_outdated(self):
+        """Test that reviews are marked as outdated correctly
+           and that updatingt the review does the right thing"""
+        review = self._save_review(None)
+        self.assertTrue(review.is_current())
+        # Change the talk time
```

Comment:
I think "talk time" here is misleading. "Update the talk", "Create a new talk revision"?

Reply:
Comments reworded to make the logic being tested clearer